src/pages/Applicant/OfferDetail/Detail/container.tsx

```diff
@@ -31,14 +30,18 @@ const DetailContainer: FunctionComponent = () => {
 
   offer.sections = sortBy(offer.sections, ["displayOrder"]);
 
-  const onSubmit = async (offerUuid: string) => {
-    await saveJobApplication({ variables: { offerUuid } });
-    history.push(RoutesBuilder.applicant.home);
+  const apply = async (offerUuid: string) => {
+    const { error } = await saveJobApplication({
+      variables: { offerUuid },
+      handlers: { JobApplicationAlreadyExistsError: () => alert(translations.data.alreadyApplied) },
+      update: cache => cache.writeData({ id: `Offer_${offerUuid}`, data: { hasApplied: true } })
+    });
+    if (!error) alert(translations.data.applySuccess);
```

Comment:
que onda este alert?

Reply:
en un issue futuro se van a incluir popups, es algo que seba había visto y terminamos pateando cuando repriorizamos en la call con beiró. prefiero de momento hacer alerts para despues hacer un find and replace global de alert() por mostrar un popup